examples/02-plot/sphere_eversion.py

```diff
@@ -0,0 +1,262 @@
+    )
+    x /= np.cos(theta)**n
+    y = (
+        t * (1 - lamda + lamda * np.cos(theta)**n) * np.sin(phi)
+        + lamda * w * np.sin(theta) * np.cos(phi)
+    )
+    y /= np.cos(theta) ** n
+    z = (
+        lamda * (
+            (w * np.sin(theta) * (np.sin(n * phi) - q * t)) / np.cos(theta)**n
+            - t / n * np.cos(n * phi)
+        )
+        - (1 - lamda) * eta**(1 + kappa) * t * abs(t)**(2 * kappa)
+            * np.sin(theta) / np.cos(theta)**(2 * n)  # noqa
+    )
+    # fmt: on
+
+    # apply Eq. (10)
+    denominator = x**2 + y**2
+    x2 = x * eta**kappa / denominator ** (1 - kappa)
+    y2 = y * eta**kappa / denominator ** (1 - kappa)
+    z2 = -z / eta / denominator
+    return x2, y2, z2
+
+
+###############################################################################
+# Now chain the functions by performing the process in Table 1 of the paper.
+# Start from the bottom for ``t = -1/Q``, keep stepping up, linearly changing
+# parameters that change from row to row, then at the top go from ``t = -1/Q``
+# to ``t = 1/Q``, then go back from top to bottom. Save each frame to a GIF.
+#
+# We make good use of the ``backface_params`` keyword parameter of
+# :func:`pyvista.Plotter.add_mesh`, allowing us to plot the inside and the
+# outside with different colors.
+
+# plot options to use for each frame
+opts = dict(
+    color='aquamarine',
+    specular=1.0,
+    specular_power=50.0,
+    backface_params=dict(color='forestgreen'),
+    smooth_shading=True,
+    reset_camera=True,
+)
+
+# use a small figure window to reduce the size of the GIF
+plotter = pv.Plotter(window_size=(300, 300))
+plotter.open_gif('sphere_eversion.gif')
+
+
+def save_frame(x, y, z):
+    """Helper to generate and store a frame of the eversion."""
+    plotter.clear()
+    plotter.enable_lightkit()
+    plotter.add_mesh(pv.StructuredGrid(x, y, z), **opts)
+    plotter.write_frame()
+
+
+# initial parameters, will be updated
+t = -1 / Q
+q = Q
+p = xi = alpha = 0
+eta = 1
+
+# sphere -> inverted wormhole
+theta, phi = np.mgrid[-np.pi / 2 : np.pi / 2 : 200j, -np.pi : np.pi : 400j]
+h, phi = sphere_to_cylinder(theta, phi)
+for lamda in np.linspace(0, 1, n_steps, endpoint=False):
+    x2, y2, z2 = unfold_sphere(theta, phi, t, q, eta, lamda)
+    save_frame(x2, y2, z2)
+
+# inverted wormhole -> unfolded wormhole
+x, y, z = cylinder_to_wormhole(h, phi, t, p, q)
+xis = np.linspace(0, 1, n_steps)
+alphas = np.linspace(0, alpha_final, n_steps)
+etas = np.linspace(1, eta_final, n_steps)
+for xi, alpha, eta in zip(xis, alphas, etas):
+    x2, y2, z2 = close_wormhole(x, y, z, eta, xi, alpha)
+    save_frame(x2, y2, z2)
+
+# unfolded wormhole -> closed wormhole
+for q in np.linspace(Q, 0, n_steps):
+    p = 1 - abs(q * t)
+    x, y, z = cylinder_to_wormhole(h, phi, t, p, q)
+    x2, y2, z2 = close_wormhole(x, y, z, eta, xi, alpha)
+    save_frame(x2, y2, z2)
+
+# closed wormhole turned inside out (flip sign of time)
+# unfolded wormhole -> closed wormhole
```

Comment:
```suggestion
# closed wormhole turned inside out (a flip sign of time)
# unfolded wormhole -> closed wormhole
```

Reply:
"Flip" is a verb here originally, so this doesn't work.

Reply:
I see, that makes sense.